app/services/arcgis_api/geocoder.rb

```diff
@@ -41,6 +39,13 @@ class Geocoder
       :SingleLine, # Unvalidated address-like text string used to search for geocoded addresses
     ]
 
+    attr_accessor :token_keeper, :connection_factory
```

Comment:
We should keep the `token_keeper` and `connection_factory` accessors private.

Reply:
corrected.